runtime/parachains/src/inclusion_inherent.rs

```diff
@@ -127,6 +129,47 @@ decl_module! {
 	}
 }
 
+/// Limit the number of backed candidates processed in order to stay within block weight limits.
+///
+/// Use a configured assumption about the weight required to process a backed candidate and the
+/// current block weight as of the execution of this function to ensure that we don't overload
+/// the block with candidate processing.
+///
+/// If the backed candidates exceed the available block weight remaining, then skips all of them.
+/// This is somewhat less desirable than attempting to fit some of them, but is more fair in the
+/// even that we can't trust the provisioner to provide a fair / random ordering of candidates.
```

Comment:
I guess we don't want any logic, i.e. deadline + binpacking combo algo come to mind.

Reply:
I don't know what a fair algorithm would be for the candidates, given the information we have available. Adding one would be a reasonable new task, but I don't think it fits within the scope of this particular issue.